shared/gh/js/utils/gh.utils.orgunits.js

```diff
@@ -0,0 +1,167 @@
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['exports'], function(exports) {
+
+    // Cache the tripos data
+    var _triposData = {};
+
+
+    //////////////
+    //  SERIES  //
+    //////////////
+
+    /**
+     * Decorate a borrowed series with its parent info
+     *
+     * @param  {Object[]}    series    A collection of series
+     */
+    var decorateBorrowedSeriesWithParentInfo = exports.decorateBorrowedSeriesWithParentInfo = function(series) {
+        _.each(series, function(serie) {
+
+            // Only decorate borrowed series
+            /* istanbul ignore else */
+            if (serie.borrowedFrom) {
+                _.each(_triposData, function(orgUnitType) {
+
+                    // Add the parent info to the organisational unit, if any
+                    var parent = _.find(orgUnitType, function(orgUnit) { return orgUnit.id === serie.borrowedFrom.ParentId; });
+                    /* istanbul ignore next */
+                    if (parent) {
+                        serie.borrowedFrom.Parent = parent;
+                    }
+                });
+            }
+        });
+    };
+
+
+    ////////////////
+    //  TRIPOSES  //
+    ////////////////
+
+    /**
+     * Decorate an organisational unit with its parent info
+     *
+     * @param  {Object}    orgUnit    The organisational unit that needs to be decorated with his parent info
+     */
+    var addParentInfoToOrgUnit = exports.addParentInfoToOrgUnit = function(orgUnit) {
+        if (orgUnit.ParentId) {
+            _.each(_triposData, function(_orgUnitType) {
+
+                // Add the parent info to the organisational unit, if any
+                var parent = _.find(_orgUnitType, function(_orgUnit) { return _orgUnit.id === orgUnit.ParentId; });
+                if (parent) {
+                    orgUnit.Parent = parent;
+                }
+            });
+        }
+    };
+
+    /**
+     * Return the tripos structure
+     *
+     * @param  {Function}    callback             Standard callback function
+     * @param  {Object}      callback.err         Error object containing the error code and error message
+     * @param  {Object}      callback.response    The tripos structure
+     */
+    var getTriposStructure = exports.getTriposStructure = function(appId, callback) {
+        if (!_.isFunction(callback)) {
+            throw new Error('An invalid value for callback was provided');
+        } else if (appId && !_.isNumber(appId)) {
+            throw new Error('An invalid value for appId was provided');
+        }
+
+        var core = require('gh.core');
+        /* istanbul ignore next */
```

Comment:
Can you test this? Should just be a case of leaving out the appId when calling the function and checking that the proper things return.